src/data_model/eng2math.jl

```diff
@@ -1,52 +1,176 @@
+            data_math["curve"] = Dict{String,Any}()
+        end
+        for (name,eng_obj) in get(data_eng["protection"],"curves",Dict())
+            math_obj = _PMD._init_math_obj("curve",name,eng_obj,length(data_math["curve"])+1;pass_props=pass_props)
+            data_math["curve"]["$(math_obj["index"])"] = math_obj
+            push!(data_math["map"], Dict{String,Any}(
+                "from" => name,
+                "to" => "curve.$(math_obj["index"])",
+                "unmap_function" => "_map_math2eng_protection!",
+                "apply_to_subnetworks" => true,
+                )
+            )
+            curve_map["$name"] = length(data_math["curve"])
+        end
+    end
+
+    if "fuses" in keys(data_eng["protection"])
+        pass_props = ["min_melt_curve","max_clear_curve","phase"]
+        if !haskey(data_math,"fuse")
+            data_math["fuse"] = Dict{String,Any}()
+        end
+        for (line_name,fuse_dict) in get(data_eng["protection"],"fuses",Dict())
+            for (name,eng_obj) in fuse_dict
+                math_obj = _PMD._init_math_obj("fuse",name,eng_obj,length(data_math["fuse"])+1;pass_props=pass_props)
+                math_obj["element"] = line_name
+                math_obj["element_enum"] = map_dict["branch"]["$line_name"]
+                if (typeof(math_obj["max_clear_curve"]) == String) || (typeof(math_obj["max_clear_curve"]) == SubString{String})
+                    math_obj["max_clear_curve_enum"] = curve_map["$(math_obj["max_clear_curve"])"]
+                end
+                if (typeof(math_obj["min_melt_curve"]) == String) || (typeof(math_obj["min_melt_curve"]) == SubString{String})
+                    math_obj["min_melt_curve_enum"] = curve_map["$(math_obj["min_melt_curve"])"]
+                end
+                data_math["fuse"]["$(math_obj["index"])"] = math_obj
+                push!(data_math["map"], Dict{String,Any}(
+                    "from" => name,
+                    "to" => "fuse.$(math_obj["index"])",
+                    "unmap_function" => "_map_math2eng_protection!",
+                    "apply_to_subnetworks" => true,
+                    )
+                )
+            end
+        end
+    end
+
+
+end
+
+
+"field/values to passthrough from the ENGINEERING to MATHEMATICAL data models"
+const _pmp_eng2math_passthrough = Dict{String,Vector{String}}(
+        "generator" => String["zr", "zx", "grid_forming"],
+        "solar" => String["i_max", "solar_max", "kva", "pf", "grid_forming"],
+        "voltage_source" => String["zr", "zx", "grid_forming"],
+    )
+
+
+"custom version of `transform_data_model` from PowerModelsDistribution for easy model transformation"
+transform_data_model(
+    data::Dict{String,<:Any};
+    eng2math_extensions::Vector{<:Function}=Function[],
+    make_pu_extensions::Vector{<:Function}=Function[],
+    kwargs...) = _PMD.transform_data_model(
+        data;
+        eng2math_extensions=[_eng2math_fault!, eng2math_extensions...],

```

Comment:
I think `_eng2math_protection!` needs to be added here

Reply:
Added